### ISSUE TITLE
[Feature Request] Deploy WASM version to GitHub Pages

### ISSUE THREAD

**Issue by AlexHedley** [Blazored/TextEditor#78]
Is the `ci-main.yml` being used?

- https://github.com/Blazored/TextEditor/blob/main/.github/workflows/ci-main.yml

Runs

- https://github.com/Blazored/TextEditor/actions/workflows/ci-main.yml

Deployed version

- https://blazored.github.io/TextEditor

---

Like other **Blazored** apps have some docs

- https://blazored.github.io/Modal/

**Comment by ADefWebserver** [comment]
@chrissainty set that up. Let's see what he says.

**Comment by AlexHedley** [comment]
Deploy failed ❌

**v1.1.1 🌈 #12**
https://github.com/Blazored/TextEditor/actions/runs/12522485891

> error: Response status code does not indicate success: 403 (The specified API key is invalid, has expired, or does not have permission to access the specified package.).

**Comment by ADefWebserver** [comment]
Yeah I saw that. I reached out to @chrissainty for assistance. I am actually not an owner of the nuget package.

**Comment by ADefWebserver** [comment]
I'm still trying to contact @chrissainty :( If I can't get a hold of him,  I will have to make a new Nuget package and lose all the history of the last one (sigh).

**Comment by AlexHedley** [comment]
Happy to wait, hopefully they can update the key or make you joint owner in NuGet or add you to the GitHub Org.

**Comment by ADefWebserver** [comment]
@AlexHedley - @chrissainty was just on vacation. The deployment has happened :) Let's see what he says about this issue when he has a chance to look at it. Thanks

**Comment by AlexHedley** [comment]
Holiday? Who has them! :p
Hope you managed to recharge @chrissainty!

**Comment by AlexHedley** [comment]
Does GitHub Pages just need configuring in Settings?
Looks like the `gh-pages` branch has the demo site but there isn’t a _pages-build-deployment_ action triggering.

- https://blazored.github.io/TextEditor

**Comment by ADefWebserver** [comment]
@AlexHedley - I just made this change. I'm not sure what to do at this point to lick it off... 
![image](https://github.com/user-attachments/assets/b3b3c7fb-fc9b-44e7-b113-f5fb4ed328b6)

**Comment by AlexHedley** [comment]
Swap `main` to `gh-pages`, click **Save** and it should start the action :)

**Comment by ADefWebserver** [comment]
Ok did it. Let me know what happens...

**Comment by AlexHedley** [comment]
New Action/Workflow triggered:

https://github.com/Blazored/TextEditor/actions/workflows/pages/pages-build-deployment

Getting a _loading_ page now on

- https://blazored.github.io/TextEditor/

I need to switch to a desktop to check what's going on but we're one step closer and nearly there, I think.

There's probably a certain file not loading, normally the `base` re-write is enough.

**Comment by chrissainty** [comment]
Check out the index page for Blazor Local Storage. The WebAssembly project is deployed to a GitHub pages site. You can find what you need there

https://github.com/Blazored/LocalStorage/blob/main/samples/BlazorWebAssembly/wwwroot/index.html

**Comment by ADefWebserver** [comment]
@chrissainty Thanks! I will look at this tomorrow. One question, what is the Pages configuration in Settings? I did not see a way to point to the "Samples/" directory. Only the root and docs.

**Comment by AlexHedley** [comment]
For comparison, the GHA will build the app then use the output to deploy to the `gh-pages` branch so it’s at the root of the folder, that is then deployed so `Samples/` folder would no longer exist.

Then when GitHub creates a URL it becomes ORG/REPONAME so another task is ran to update the base path from `/` to `/REPONAME`, I usually just `sed` it but looks like SteveS created an action to do it a while ago I didn’t know about.

There’s a great write up here:
https://swimburger.net/blog/dotnet/how-to-deploy-aspnet-blazor-webassembly-to-github-pages

You also need a 404 file to redirect because of routing.

**TextEditor**
https://github.com/Blazored/TextEditor/tree/gh-pages

**LocalStorage**
https://github.com/Blazored/LocalStorage/tree/gh-pages

**Comment by ADefWebserver** [comment]
@AlexHedley - I took some time to look at this today and I'm lost :( If there is a PR you could make I will merge it :)

**Comment by AlexHedley** [comment]
Sure thing

**Comment by AlexHedley** [comment]
- https://github.com/Blazored/TextEditor/pull/84

**Comment by ADefWebserver** [comment]
Merged. Thank you

**Comment by ADefWebserver** [comment]
Finally! [https://blazored.github.io/TextEditor/](https://blazored.github.io/TextEditor/) Thanks @AlexHedley !

**Comment by AlexHedley** [comment]
Looks great :-)

You can also add a link in the right navigation under tags and deployments/packages etc
There's an option to use GitHub Pages url via a checkbox once you go into Settings ⚙️

**Comment by ADefWebserver** [comment]
@AlexHedley Done!

**Comment by AlexHedley** [comment]
Fab, I love it when a plan comes together :-)
